Tests/AtomicWrapperSpec.swift

```diff
@@ -0,0 +1,185 @@
+import Quick
+import Nimble
+@testable import RInAppMessaging
+
+class AtomicWrapperSpec: QuickSpec {
+
+    @AtomicGetSet var atomicArray = [String]()
+
+    override func spec() {
+
+        struct DelayedValue<T> {
+            private let queue = DispatchQueue(label: "DelayedQueue")
+            private let delaySeconds = 0.1
+
+            let value: T
+
+            init(_ value: T) {
+                self.value = value
+            }
+
+            func get() -> T {
+                queue.sync {
+                    usleep(useconds_t(delaySeconds * Double(USEC_PER_SEC)))
+                    return value
+                }
+            }
+        }
+
+        describe("AtomicGetSet property wrapper") {
+
+            let queueA = DispatchQueue(label: "QueueA")
+            let queueB = DispatchQueue(label: "QueueB")
+
+            beforeEach {
+                self.atomicArray = []
+            }
+
+            it("will not crash when two threads access the same value at the same time (get)") {
+                let dispatchGroup = DispatchGroup()
+                dispatchGroup.enter()
+                dispatchGroup.enter()
+
+                queueA.async {
+                    for _ in (1...1_000_000) {
+                        _ = self.atomicArray
+                    }
+                    dispatchGroup.leave()
+                }
+                queueB.async {
+                    for _ in (1...1_000_000) {
+                        _ = self.atomicArray
+                    }
+                    dispatchGroup.leave()
+                }
+                dispatchGroup.wait()
+            }
+
```

Comment:
Should there also be a reading/writing thread safety test?

Reply:
Like one thread reads and the other one writes the value? Yeah I can add that.